src/main/java/jalf/compiler/Cog.java

```diff
@@ -157,4 +158,17 @@ public Cog union(Union union, Cog right) {
         return new BaseCog(union, supplier);
     }
 
+    /** Default compilation of `minus`. */
+    public Cog minus(Minus minus, Cog right) {
+        Supplier<Stream<Tuple>> supplier = () ->{
+            Stream<Tuple> leftStream = this.stream();
+            Stream<Tuple> rightStream = right.stream();
+            List<Tuple> rightList = rightStream.collect(Collectors.toList());
+            Stream<Tuple> minusStream = leftStream
+                    .filter(x -> !rightList.contains(x));
+            return minusStream;
+        };
+        return new BaseCog(minus, supplier);
+    }
```

Comment:
@aimablehirwa  Complexity is way too high. We should at least use another data structure than an List for intermediate result. Possibly, we could use other strategies, to be thought of.